src/tracer/MARBL_tracers.F90

```diff
@@ -0,0 +1,734 @@
+!> A tracer package for tracers computed in the MARBL library
+module MARBL_tracers
+
+! This file is part of MOM6. See LICENSE.md for the license.
+
+! Currently configured for use with marbl0.36.0
+! https://github.com/marbl-ecosys/MARBL/releases/tag/marbl0.36.0
+! (clone entire repo into pkg/MARBL)
+#ifdef _USE_MARBL_TRACERS
+use MOM_coms,            only : root_PE, broadcast
+use MOM_diag_mediator,   only : diag_ctrl
+use MOM_error_handler,   only : is_root_PE, MOM_error, FATAL, WARNING
+use MOM_file_parser,     only : get_param, log_param, log_version, param_file_type
+use MOM_forcing_type,    only : forcing
+use MOM_grid,            only : ocean_grid_type
+use MOM_hor_index,       only : hor_index_type
+use MOM_io,              only : file_exists, read_data, slasher, vardesc, var_desc, query_vardesc
+use MOM_open_boundary,   only : ocean_OBC_type
+use MOM_restart,         only : query_initialized, MOM_restart_CS
+use MOM_sponge,          only : set_up_sponge_field, sponge_CS
+use MOM_time_manager,    only : time_type
+use MOM_tracer_registry, only : register_tracer, tracer_registry_type
+use MOM_tracer_diabatic, only : tracer_vertdiff, applyTracerBoundaryFluxesInOut
+use MOM_tracer_Z_init,   only : tracer_Z_init
+use MOM_unit_scaling,    only : unit_scale_type
+use MOM_variables,       only : surface, thermo_var_ptrs
+use MOM_verticalGrid,    only : verticalGrid_type
+use MOM_diag_mediator,   only : register_diag_field, post_data!, safe_alloc_ptr
+
+use MARBL_interface,              only : MARBL_interface_class
+use MARBL_interface_public_types, only : marbl_diagnostics_type, marbl_saved_state_type
+
+use coupler_types_mod,      only : coupler_type_set_data, ind_csurf
+use atmos_ocean_fluxes_mod, only : aof_set_coupler_flux
+
+implicit none ; private
+
+#include <MOM_memory.h>
+
+public register_MARBL_tracers, initialize_MARBL_tracers
+public MARBL_tracers_column_physics, MARBL_tracers_surface_state
+public MARBL_tracer_stock, MARBL_tracers_end
+
+! A note on unit descriptions in comments: MOM6 uses units that can be rescaled for dimensional
+! consistency testing. These are noted in comments with units like Z, H, L, and T, along with
+! their mks counterparts with notation like "a velocity [Z T-1 ~> m s-1]".  If the units
+! vary with the Boussinesq approximation, the Boussinesq variant is given first.
+
+!> Temporary type for diagnostic variables coming from MARBL
+!! Allocate exactly one of field_[23]d
+type, private :: temp_MARBL_diag
```

Comment:
The `private` declarations are redundant here because of line 36

Reply:
My next commit will address this sort of fix (rather than copying and pasting this comment a bunch of times, I'll just use the thumbs up reaction)